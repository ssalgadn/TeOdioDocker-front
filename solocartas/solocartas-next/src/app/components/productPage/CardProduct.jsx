'use client';

import { formatPriceCLP } from '@/utils/format';
import { useRouter } from 'next/navigation';

export default function CardProduct({ product }) {
  const router = useRouter();

  const handleClick = () => {
    router.push(`/products/${product.id}`);
  };

  return (
    <div 
      onClick={handleClick}
      className="border rounded-lg overflow-hidden shadow hover:shadow-lg transition bg-white cursor-pointer"
    >
      <img
        src={product.image}
        alt={`Carta ${product.name}`}
        className="w-full h-[320px] object-cover bg-gray-100"
      />

      <div className="p-3 flex flex-col items-start">
        <h2 className="text-base font-semibold mb-1">{product.name}</h2>
        <p className="text-green-600 font-medium mb-1">{formatPriceCLP(product.price)}</p>
        <small className="text-gray-500 text-sm">Vendido por: {product.store}</small>
      </div>
    </div>
  );
}
