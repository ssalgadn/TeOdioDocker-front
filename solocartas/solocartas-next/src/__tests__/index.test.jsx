import { render, screen, fireEvent } from '@testing-library/react';
import CardProduct from '@/app/components/productPage/CardProduct';
import { formatPriceCLP } from '@/utils/format';
import { axe } from 'jest-axe';

jest.mock('next/image', () => (props) => {
  // eslint-disable-next-line @next/next/no-img-element
  return <img {...props} alt={props.alt} />;
});

describe('utils/format', () => {
  it('formats number to Chilean peso', () => {
    expect(formatPriceCLP(1234)).toBe('$1.234');
  });
});

describe('CardProduct component', () => {
  const product = {
    id: 1,
    name: 'Charizard VMAX',
    img_url: '/charizard.png',
    min_price: 15000,
  };

  it('renders product data correctly', () => {
    render(<CardProduct product={product} />);

    expect(screen.getByText(product.name)).toBeInTheDocument();
    expect(screen.getByText(formatPriceCLP(product.min_price))).toBeInTheDocument();
    expect(screen.getByAltText(`Carta ${product.name}`)).toBeInTheDocument();
  });

  it('navigates to product detail when clicked', () => {
    const mockPush = jest.fn();
    jest.spyOn(require('next/navigation'), 'useRouter').mockImplementation(() => ({ push: mockPush }));
    render(<CardProduct product={product} />);

    fireEvent.click(screen.getByAltText(`Carta ${product.name}`));
    expect(mockPush).toHaveBeenCalledWith(`/products/${product.id}`);
  });

  it('is accessible', async () => {
    const { container } = render(<CardProduct product={product} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<CardProduct product={product} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
