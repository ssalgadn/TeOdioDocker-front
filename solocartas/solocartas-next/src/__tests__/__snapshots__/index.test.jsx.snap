// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardProduct component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="border dark:border-gray-700 rounded-lg overflow-hidden shadow hover:shadow-lg transition bg-white dark:bg-slate-800 cursor-pointer"
  >
    <img
      alt="Carta Charizard VMAX"
      class="w-full h-[320px] object-cover bg-gray-100 dark:bg-slate-700"
      src="/charizard.png"
    />
    <div
      class="p-3 flex flex-col items-start"
    >
      <h2
        class="text-base font-semibold mb-1 dark:text-gray-100"
      >
        Charizard VMAX
      </h2>
      <p
        class="text-green-600 font-medium mb-1"
      >
        $15.000
      </p>
    </div>
  </div>
</DocumentFragment>
`;
